### PR TITLE
Scoutsuite automation test

### PR DESCRIPTION
**Pull request by gcasilva**

### CODE REVIEW

File: .project_automation/functional_tests/entrypoint.sh [hunk 1]
```diff
@@ -23,13 +23,46 @@ cleanup_all_regions() {
...
+        echo "Scoutsuite report contains security issues. Details below or check the S3 bucket named scoutsuite-results-aws-AWS-ACCOUNT-ID in the test account."
+        
+        # Retrieve the AWS account ID and store it in a variable
+        AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query "Account" --output text)
+        
+        # Path to the scoutsuite JSON file
+        FILE="./scoutsuite-report/scoutsuite-results/scoutsuite_results_aws-$AWS_ACCOUNT_ID.js"
+
+        # Extract and filter data
+        tail -n +2 scoutsuite-report/scoutsuite-results/scoutsuite_results_aws-565009154835.js | jq '.'
```
Comment (kkvinjam) [inline .project_automation/functional_tests/entrypoint.sh:52]: AWS account-id should not be hard-coded.

File: .project_automation/functional_tests/entrypoint.sh [hunk 2]
```diff
@@ -23,13 +23,46 @@ cleanup_all_regions() {
...
+process_scoutsuite_report() {
+    # Check Scoutsuite security scan result for Danger level findings (Non-0 exit code)
+    scoutsuite_sysout_result=$(cat scoutsuite_sysout.txt)
+    #rm scoutsuite_sysout.txt
+    if [ "$scoutsuite_sysout_result" -ne 0 ]; then
+        # The value is non-zero, indicating Scoutsuite report needs to be checked for security issues
+        echo "Scoutsuite report contains security issues. Details below or check the S3 bucket named scoutsuite-results-aws-AWS-ACCOUNT-ID in the test account."
+        
+        # Retrieve the AWS account ID and store it in a variable
+        AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query "Account" --output text)
```
Comment (kkvinjam) [inline .project_automation/functional_tests/entrypoint.sh:46]: Can we move this to the top outside the function? I believe there is no need to run this command multiple times.

File: .project_automation/functional_tests/entrypoint.sh [hunk 3]
```diff
@@ -23,13 +23,46 @@ cleanup_all_regions() {
...
+    scout aws -r us-east-1 --ruleset .project_automation/functional_tests/abi-scoutsuite-custom-ruleset.json --no-browser --max-rate 5 --max-workers 5
+    # Upload Scoutsuite security scan results to S3 bucket named scoutsuite-results-aws-AWS-ACCOUNT-ID
+    python3 .project_automation/functional_tests/process-scoutsuite-report.py
+    # Delete taskcat e2e test resources
+    taskcat test clean ALL
+}
+
+process_scoutsuite_report() {
+    # Check Scoutsuite security scan result for Danger level findings (Non-0 exit code)
+    scoutsuite_sysout_result=$(cat scoutsuite_sysout.txt)
```
Comment (kkvinjam) [inline .project_automation/functional_tests/entrypoint.sh:39]: How is deletion between the steps handled?

File: .project_automation/functional_tests/entrypoint.sh (hunk 2 above)
Comment (gcasilva) [reply .project_automation/functional_tests/entrypoint.sh]: Moved to the top outside of the function.

**Comment by gcasilva** [comment]
@kkvinjam Made changes to create a folder inside .project_automation/functional_tests for all files related to scoutsuite, so its cleaner. Also made all the changes we talked about. E2E tests completed successfully. Please review whenever you can.

**Comment by gcasilva** [comment]
@kkvinjam Please approve changes when possible.

**Review by kkvinjam — APPROVED** [review]

**Pull request merged**